py/h2o_wave/h2o_wave/server.py

```diff
@@ -75,13 +75,27 @@ def __init__(self, username: str, subject: str, access_token: str, refresh_token
         """The refresh token of the user."""
         self._session_id = session_id
         """Session identifier. Do not access, internal use only."""
-    
+
     async def ensure_fresh_token(self) -> Optional[str]:
         """
         Explicitly refresh OIDC tokens when needed, e.g. during long-running background jobs.
         """
         async with httpx.AsyncClient(auth=(_config.hub_access_key_id, _config.hub_access_key_secret), verify=False) as http:
-            res = await http.get(_config.hub_address + '_auth/refresh', headers={'Wave-Session-ID': self._session_id}) 
+            res = await http.get(_config.hub_address + '_auth/refresh', headers={'Wave-Session-ID': self._session_id})
+
+            access_token = res.headers.get('Wave-Access-Token', None)
+            refresh_token = res.headers.get('Wave-Refresh-Token', None)
+            if access_token and refresh_token:
+                self.access_token = access_token
+                self.refresh_token = refresh_token
+            return access_token
```

Comment:
nit: this can be reused as there's no async code.

Reply:
Fair point.